app/controllers/things_controller.rb

```diff
@@ -3,10 +3,21 @@ class ThingsController < ApplicationController
 
   def show
     @things = Thing.find_closest(params[:lat], params[:lng], params[:limit] || 10)
-    if @things.blank?
-      render(json: {errors: {address: [t('errors.not_found', thing: t('defaults.thing'))]}}, status: 404)
+    if @things
+      @things.each do |t|
+        t.owned_by_you = check_owner(t)
+      end
+      render(json: @things.to_json(methods: :owned_by_you))
     else
-      respond_with @things
+      render(json: {'errors' => {'address' => [t('errors.not_found', thing: t('defaults.thing'))]}}, status: 404)
+    end
+  end
+
+  def check_owner(thing)
```

Comment:
I think this method better belongs on the model, something like:

```
def is_owned_by?(owner)
  return false unless owner

  return self.user.id == owner.id
end
```

Just to adhere to the "fat model, skinny controller" pattern.